### DISCUSSION TITLE
Ordering resources by name

### DISCUSSION THREAD

**Discussion by spacerobotTR** [excid3/madmin#61] [Q&A]
What is the best way to order a resource on the index action by a certain field instead of the default id ordering?

**Comment by excid3** [comment]
Sortable columns is still to be added at the moment.

**Reply by excid3** [reply]
Added! 

Felt like working on them today. 😜

**Comment by spacerobotTR** [comment]
Great. Thanks!

**Reply by spacerobotTR** [reply]
To use it is the format like this?   
attribute :groupname, sort: true, direction: 'asc' 
or am I doing this wrong?

**Reply by excid3** [reply]
Whoops, yeah I just added sorting in general to the index. We'll need to add a setting to the resource to override the default column and direction still. Should be quick to add though.

**Reply by excid3** [reply]
Added!

```ruby
class PostResource
  def self.default_sort_column
    "title"
  end

  def self.default_sort_direction
    "asc"
  end
end
```

**Reply by spacerobotTR** [reply]
Awesome! I still haven't gotten it to work. Updated the gem to version 1.20. Here is my resource file. Anything else I am missing?

```
class CardholdergroupResource < Madmin::Resource
  # Attributes
  attribute :id, form: false
  attribute :groupname
  attribute :description
  attribute :grouptype
  attribute :created_at, form: false, index:false
  attribute :updated_at, form: false, index:false

  def self.default_sort_column
  "groupname"
  end

  def self.default_sort_direction
    "asc"
  end

end
```
